src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/defaultsite/Controller/DefaultController.php

```diff
@@ -2,22 +2,44 @@
 
 namespace {{ namespace }}\Controller;
 
-use \Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
-use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
 
 class DefaultController extends Controller
 {
     {% if multilanguage %}
-
     /**
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse
      * @Route("/")
+     * @param Request $request
+     * @return RedirectResponse
+     * @throws \InvalidArgumentException
+     */
+    public function indexAction(Request $request)
+    {
+        return new RedirectResponse(
+            $this->generateUrl(
+                '_slug',
+                [
+                    'url'       => '',
+                    '_locale'   => $this->getLocale($request)
+                ]
+            )
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @return string
+     * @throws \Symfony\Component\DependencyInjection\Exception\InvalidArgumentException
      */
-    public function indexAction()
+    private function getLocale(Request $request)
     {
-        return new RedirectResponse($this->generateUrl('_slug', array('url'=>'', '_locale'=>$this->container->getParameter('locale'))));
+        $locales = array_filter(
+            explode('|', $this->container->getParameter('requiredlocales'))
+        );
+        return $request->getPreferredLanguage($locales);
```

Comment:
The method getPreferredLanguages documentation states the method could return null.
http://api.symfony.com/3.1/Symfony/Component/HttpFoundation/Request.html#method_getPreferredLanguage

 Looking at the implementation itself I don't think it would ever happen in our use case but I do think it would be best to add a null check to this code. This way whenever the implementation changes we can be sure the functionality won't break. 
Seems like a good idea to check for null and return $this->container->getParameter('locale') if this happens.

Reply:
@Numkil, I think is not necessary because if You look at KunstmaanStandardEdition in routing.yml, every single route have a requirements. 
```
# KunstmaanAdminBundle
KunstmaanAdminBundle:
    resource: "@KunstmaanAdminBundle/Resources/config/routing.yml"
    prefix:   /{_locale}/
    requirements:
        _locale: "%requiredlocales%"

# KunstmaanAdminListBundle
KunstmaanAdminListBundle:
    resource: "@KunstmaanAdminListBundle/Resources/config/routing.yml"
    prefix:   /{_locale}/
    requirements:
        _locale: "%requiredlocales%"
```
In conclusion, if `requiredlocales` is null we give a exception  `InvalidParameterException`.

Reply:
@Numkil, I think is not necessary, if You look at KunstmaanStandardBundle in routing.yml every single routes has a required parameter `_locale` provided by `%requiredlocales%`. If `requiredlocales` is null then Symfony give a exception from routing.

```
# KunstmaanAdminListBundle
KunstmaanAdminListBundle:
    resource: "@KunstmaanAdminListBundle/Resources/config/routing.yml"
    prefix:   /{_locale}/
    requirements:
        _locale: "%requiredlocales%"

# KunstmaanNodeBundle
KunstmaanNodeBundle:
    resource: "@KunstmaanNodeBundle/Resources/config/routing.yml"
    prefix:   /{_locale}/
    requirements:
        _locale: "%requiredlocales%"

```

Reply:
@cv65kr 
Ok that makes sense.